src/Infrastructure.Dapper/Repositories/ProviderOrganizationRepository.cs

```diff
@@ -18,6 +19,22 @@ public ProviderOrganizationRepository(string connectionString, string readOnlyCo
         : base(connectionString, readOnlyConnectionString)
     { }
 
+    public async Task<ICollection<ProviderOrganization>> CreateWithManyOrganizations(ProviderOrganization providerOrganization, IEnumerable<Guid> organizationIds)
+    {
+        var objWithOrganizationIds = JsonSerializer.Deserialize<ProviderOrganizationWithOrganizations>(
+            JsonSerializer.Serialize(providerOrganization));
+        objWithOrganizationIds.OrganizationIds = organizationIds.ToGuidIdArrayTVP();
```

Comment:
I don't really understand what's going on here to be honest. We serialize & then deserialize `ProviderOrganization` into a `ProviderOrganizationWithOrganizations`, which is the same except it has a 1-to-many mapping done via a `DataTable` property. Then inside the sproc we use this to populate an in-memory table, then use that in memory table to populate insert the changes into the database. It all seems very convoluted.

I don't have another suggestion off the top of my head, but is there a simpler way? How do our other sprocs do this? Let me know if you want me to come up with a suggested code snippet.

Reply:
I looked at what the other methods were using for inserting multiple records and most of them use `SqlBulkCopy` and in one case it used a `User Defined Type` to pass multiple entries to a stored procedure.
I've implemented a solution using `SqlBulkCopy` because it seemed more direct to me but if you think the other way is better then I'm happy to change it.

Reply:
This looks a lot better to me and makes more sense. Bonus points that we don't have to add a sproc for it. I'm still going to ask for a second review though, just on the database stuff, because I'm less familiar with it.